lib/popcorn/pop_json.nit

```diff
@@ -0,0 +1,196 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Copyright 2017 Alexandre Terrasa <alexandre@moz-code.org>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Introduce useful services for JSON REST API handlers.
+#
+# Validation and Deserialization of request bodies:
+#
+# ~~~nit
+# class MyJsonHandler
+#	super Handler
+#
+#	# Validator used do validate the body
+#	redef var validator = new MyFormValidator
+#
+#	# Define the kind of objects expected by the deserialization process
+#	redef type BODY: MyForm
+#
+#	# Instanciate the object from a deserializer
+#	redef fun new_body_object(d) do return new MyForm.from_deserializer(d)
+#
+#	redef fun post(req, res) do
+#		var post = validate_body(req, res)
+#		if post == null then return # Validation error: let popcorn return a HTTP 400
+#		var form = deserialize_body(req, res)
+#		if form == null then return # Deserialization error: let popcorn return a HTTP 400
```

Comment:
I'm not sure to understand the orchestration between `validate_body` and `deserialize_body`.

* `post` returned by the first call is not used by the second call.
* in my mind, shouldn't one deserialize an object before checking its validity?